### PR TITLE
Add LDAP response messages

### PR DESCRIPTION
**Pull request by seladb**
This PR is part of #1430 

It adds the following LDAP messages:
- `LdapSearchResultDoneLayer`
- `LdapModifyResponseLayer`
- `LdapAddResponseLayer`
- `LdapDeleteResponseLayer`
- `LdapModifyDNResponseLayer`
- `LdapCompareResponseLayer`

### CODE REVIEW

File: Packet++/src/LdapLayer.cpp [hunk 1]
```diff
@@ -214,6 +330,77 @@ namespace pcpp {
...
+		const std::string& matchedDN, const std::string& diagnosticMessage, const std::vector<std::string>& referral,
+		const std::vector<LdapControl>& controls)
+	{
+		Asn1EnumeratedRecord resultCodeRecord(resultCode);
+		Asn1OctetStringRecord matchedDNRecord(matchedDN);
+		Asn1OctetStringRecord diagnosticMessageRecord(diagnosticMessage);
+
+		std::vector<Asn1Record*> messageRecords = {&resultCodeRecord, &matchedDNRecord, &diagnosticMessageRecord};
+
+		std::unique_ptr<Asn1ConstructedRecord> referralRecord;
```
Comment (Dimi1010) [inline Packet++/src/LdapLayer.cpp:345]: nit: Does `referralRecord` need to exist outside the `if` scope?

**Review by Dimi1010 — APPROVED** [review]

File: Packet++/src/LdapLayer.cpp (hunk 1 above)
Comment (Dimi1010) [reply Packet++/src/LdapLayer.cpp:345]: Nvm, I see you are using the pointer for memory management and the raw pointer is used in `init`

File: Packet++/src/LdapLayer.cpp [hunk 2]
```diff
@@ -72,6 +72,110 @@ namespace pcpp {
 
 	// endregion
 
+	// region LdapResultCode
+
+	const std::unordered_map<LdapResultCode::Value, std::string, EnumClassHash<LdapResultCode::Value>> LdapResultCodeToString{
```
Comment (tigercosmos) [inline Packet++/src/LdapLayer.cpp:77]: make it static? same for the next map.

File: Packet++/src/LdapLayer.cpp [hunk 3]
```diff
@@ -214,6 +330,77 @@ namespace pcpp {
...
+
+		for (auto uriRecord : referralRecord->castAs<Asn1ConstructedRecord>()->getSubRecords())
+		{
+			result.push_back(uriRecord->castAs<Asn1OctetStringRecord>()->getValue());
+		}
+
+		return result;
+	}
+
+	std::string LdapResponseLayer::getExtendedStringInfo() const
```
Comment (tigercosmos) [inline Packet++/src/LdapLayer.cpp:398]: maybe rename to `getExtendedInfoString`?

Comment (seladb) [reply Packet++/src/LdapLayer.cpp]: I agree, updated in this commit: https://github.com/seladb/PcapPlusPlus/pull/1437/commits/c72a2af0fc7b9c3f0dbb9b03f4cf77dc75d5bc9d

File: Packet++/src/LdapLayer.cpp (hunk 2 above)
Comment (seladb) [reply Packet++/src/LdapLayer.cpp]: I'm not sure [it really matters](https://stackoverflow.com/a/3709257), but I changed it anyway in this commit: https://github.com/seladb/PcapPlusPlus/pull/1437/commits/c72a2af0fc7b9c3f0dbb9b03f4cf77dc75d5bc9d

**Comment by seladb** [comment]
@tigercosmos I addressed your comments. Can you re-review this PR?

**Pull request merged**